sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -636,22 +636,26 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
       summary = "")
   }
 
-  def intervalArithmeticOverflowError(
-      message: String,
-      hint: String = "",
-      context: QueryContext): ArithmeticException = {
-    val alternative = if (hint.nonEmpty) {
-      s" Use '$hint' to tolerate overflow and return NULL instead."
-    } else ""
+  def withTrySuggestionIntervalArithmeticOverflowError(
+      suggestedFunc: String,
+      context: QueryContext = null): ArithmeticException = {
     new SparkArithmeticException(
-      errorClass = "INTERVAL_ARITHMETIC_OVERFLOW",
-      messageParameters = Map(
-        "message" -> message,
-        "alternative" -> alternative),
+      errorClass = "INTERVAL_ARITHMETIC_OVERFLOW.WITH_TRY_SUGGESTION",
+      messageParameters = Map("functionName" -> suggestedFunc),
```

Comment:
the sugggested function name is an identifier, so, let's quote it by `toSQLId`:
```suggestion
      messageParameters = Map("functionName" -> toSQLId(suggestedFunc)),
```